contracts/0.8.9/BeaconChainDepositor.sol

```diff
@@ -73,42 +71,26 @@ contract BeaconChainDepositor {
     /// @dev computes the deposit_root_hash required by official Beacon Deposit contract
     /// @param _publicKey A BLS12-381 public key.
     /// @param _signature A BLS12-381 signature
-    function _computeDepositDataRoot(
-        bytes memory _withdrawalCredentials,
-        bytes memory _publicKey,
-        bytes memory _signature
-    ) private pure returns (bytes32) {
+    function _computeDepositDataRoot(bytes memory _withdrawalCredentials, bytes memory _publicKey, bytes memory _signature)
+        private
+        pure
+        returns (bytes32)
+    {
         // Compute deposit data root (`DepositData` hash tree root) according to deposit_contract.sol
-        bytes32 publicKeyRoot = sha256(_pad64(_publicKey));
-        bytes32 signatureRoot = sha256(
+        bytes memory sigPart1 = MemUtils.unsafeAllocateBytes(64);
+        bytes memory sigPart2 = MemUtils.unsafeAllocateBytes(SIGNATURE_LENGTH - 64);
+        MemUtils.copyBytes(_signature, sigPart1, 0, 0, 64);
+        MemUtils.copyBytes(_signature, sigPart2, 64, 0,SIGNATURE_LENGTH - 64);
+
+        bytes32 publicKeyRoot = sha256(abi.encodePacked(_publicKey, bytes16(0)));
+        bytes32 signatureRoot = sha256(abi.encodePacked(sha256(abi.encodePacked(sigPart1)), sha256(abi.encodePacked(sigPart2, bytes32(0)))));
```

Comment:
please please let's not have lines longer than 100 symbols

Reply:
take into account

Reply:
We have "printWidth": 120 for solidity.